packages/core/src/components/widgets/AltDateWidget.tsx

```diff
@@ -72,13 +81,16 @@ class AltDateWidget extends Component {
     }
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
+  shouldComponentUpdate(nextProps: WidgetProps<T, F>, nextState: any): boolean {
```

Comment:
Since you made the state type above be `DateObject` you can make `nextState: DateObject`